lib/macho/exceptions.rb

```diff
@@ -3,6 +3,10 @@ module MachO
   class MachOError < RuntimeError
   end
 
+  # A generic Mach-O error during file modification.
+  class MachOModificationError < MachOError
```

Comment:
No worries! I was also confusing myself my reusing the term in the exceptions. How do you feel about `CheckedModificationError` and `UncheckedModificationError` instead? That gets us away from the connotation of program halting and more accurately conveys the fact that we're checking and filtering subclasses of the particular error.